ansible_collections/arista/cvp/plugins/module_utils/device_tools.py

```diff
@@ -1739,6 +1780,118 @@ def reset_device(self, user_inventory: DeviceInventory):
+                                    device.fqdn
+                                )
+                            )
+                            self.__ansible.fail_json(
+                                msg="Error validation failed on device {0}".format(
+                                    device.fqdn
+                                )
+                            )
+                        else:
+                            if "result" in resp:
+                                result_data.changed = True
+                                result_data.success = True
+                            if "warnings" in resp and resp["warningCount"] > 0:
+                                err_msg = resp["warnings"]
+                                msg = f"Configlet validation failed with {err_msg}"
+                                result_data.success = False
+                                result_data.count += 1
+                                MODULE_LOGGER.error(msg)
+                                device_data["warnings"].append(
+                                    {"device": device.hostname, "warnings": err_msg}
+                                )
+                            if "errors" in resp:
+                                err_msg = resp["errors"]
+                                msg = f"Configlet validation failed with {err_msg}"
+                                result_data.success = False
+                                result_data.count += 1
+                                MODULE_LOGGER.error(msg)
+                                device_data["errors"].append(
+                                    {"device": device.hostname, "errors": err_msg}
+                                )
+                    result_data.add_entry(
+                        "{0} validated against {1}".format(
+                            device.info[self.__search_by], vc_configlet
+                        )
+                    )
+            results.append(result_data)
+        MODULE_LOGGER.debug("device_data is: {0}".format(str(device_data)))
+        if len(device_data["errors"]) > 0:
+            if len(device_data["warnings"]) == 0:  # errors but no warnings
+                if validate_mode == vldm_skip or validate_mode == vldm_warn:
+                    self.__ansible.exit_json(msg=str(device_data))
+                else:
+                    self.__ansible.fail_json(msg=str(device_data))
+            elif len(device_data["warnings"]) > 0:  # errors and warnings
+                if validate_mode == vldm_err or validate_mode == vldm_warn:
+                    self.__ansible.fail_json(msg=str(device_data))
+                if validate_mode == vldm_skip:
+                    self.__ansible.exit_json(msg=str(device_data))
+
+        # warnings but no errors
+        elif len(device_data["warnings"]) > 0 and len(device_data["errors"]) == 0:
+            if validate_mode == vldm_skip or validate_mode == vldm_err:
```

Comment:
this case is a bit peculiar to me - if I ask to stop on warnings I would expect that it stops on warning or error not only on warning - as an error is worse than a warning.

probably logically I would rewrite the all section somthing like:

```
if len(device_data["errors"]) > 0 and validate_mode in [vldm_warn, vldm_err]:
  self.__ansible.fail_json(msg=str(device_data))
elif len(device_data["warnings"]) > 0 and validate_mode == vldm_warn:
  self.__ansible.fail_json(msg=str(device_data))
else:
  self.__ansible.exit_json(msg=str(device_data))
```

Reply:
afaik that's the logic I had it initially, but I see @sugetha24 changed it, your suggestion looks good, let me rewrite that thanks!